src/System.Private.CoreLib/src/System/Threading/ThreadLocal.cs

```diff
@@ -183,23 +171,23 @@ protected virtual void Dispose(bool disposing)
         {
             int id;
 
-            using (LockHolder.Hold(s_idManager.m_lock))
+            lock (s_idManager)
```

Comment:
The `LockHolder.Hold` is a perf optimization (see 4c22bc35aca93f56c0cba333c77ae99265f09733 where it was added). Are we planning to get rid of it when the CoreCLR change ports over? Cc @AntonLapounov

Reply:
I am ok with keeping the LockHolder on very hot paths with ifdefs. This is not one of them.

Everything in corefx and large part of corelib is on the `lock` plan, so the `lock` keyword has to work pretty well. One more or less does not really make a difference. If we wanted to use this everywhere, we would need to turn this into a proper public feature: introduce a public Lock type and tech C# compiler how to use it with `lock` keyword.

Reply:
I agree with @jkotas.  It's OK here, however, please evaluate performance impact before removing `LockHolder` on hot paths.